modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/DynamicFilter.js

```diff
@@ -0,0 +1,281 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayMultiSelect from '@clayui/multi-select';
+import {fetch, openToast} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+
+const getSelectedItemsLabel = ({selectedData}) => {
+	return selectedData.map((item) => item.label).join(', ');
+};
+
+const getOdataString = ({id, selectedData}) => {
+	return `${id} in (${selectedData
+		.map((item) => `'${item.value}'`)
+		.join(',')})`;
+};
+
+async function getPicklist(erc) {
+	const response = await fetch(
+		`/o/headless-admin-list-type/v1.0/list-type-definitions/by-external-reference-code/${erc}`
+	);
+
+	if (!response.ok) {
+		openToast({
+			message: Liferay.Language.get('your-request-failed-to-complete'),
+			type: 'danger',
+		});
+
+		return [];
+	}
+
+	return await response.json();
+}
+
+const DynamicFilter = ({
+	id,
+	listTypeDefinitionId,
+	multiple,
+	preselectedValues,
+	selectedData,
+	setFilter,
```

Comment:
I would rename this to `onFIlterChange`. It's more react-y. `set*` is generally reserved for when you are using useState directly

Reply:
Mmm, that's the `setFilter` API we discussed about long ago. It's a callback into the filters UI and it's setting the filter flags, data, odata string, etc., that's why we named it like that instead of following the React pattern (because that would make our API look very awkward, as well as the code that uses it).

Reply:
Oh, I'm now thinking that this is not a CX, so my comment doesn't apply. 

However I copied that code from other non-CX filter, so I'm not sure why it was called `setFilter` in first place :thinking:

Reply:
Oh yeah, I vaguely remember this now.. I think it stems from an awkwardness of creating some props in a java class and passing it through to a react component. Because is there even a parent component to this?